test/parallel/test-http-mutable-headers.js

```diff
@@ -34,15 +34,35 @@ const s = http.createServer(common.mustCall((req, res) => {
         res.removeHeader();
       }, /^TypeError: "name" argument must be a string$/);
 
+      const arrayValues = [1, 2, 3];
       res.setHeader('x-test-header', 'testing');
       res.setHeader('X-TEST-HEADER2', 'testing');
       res.setHeader('set-cookie', cookies);
-      res.setHeader('x-test-array-header', [1, 2, 3]);
+      res.setHeader('x-test-array-header', arrayValues);
 
       assert.strictEqual(res.getHeader('x-test-header'), 'testing');
       assert.strictEqual(res.getHeader('x-test-header2'), 'testing');
 
+      const headersCopy = res.getHeaders();
+      // eslint-disable-next-line no-restricted-properties
+      assert.deepEqual(headersCopy, {
+        'x-test-header': 'testing',
+        'x-test-header2': 'testing',
+        'set-cookie': cookies,
+        'x-test-array-header': arrayValues
+      });
+      assert.notStrictEqual(headersCopy['set-cookie'], cookies);
+      assert.notStrictEqual(headersCopy['x-test-array-header'], arrayValues);
+
+      assert.deepStrictEqual(res.getHeaderNames(),
+                             ['x-test-header', 'x-test-header2',
+                              'set-cookie', 'x-test-array-header']);
+
+      assert.strictEqual(res.hasHeader('x-test-header2'), true);
```

Comment:
nit: being pedantic here but it would be good to expand the test to include mixed case header name (to ensure the `toLowerCase`) and passing non-string inputs to test the error checking.